src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts

```diff
@@ -98,62 +127,66 @@ export class RuntimeCodeMigration extends Migration<ComponentMigrator[], Schemat
-      const trimmedNodeText = nodeText.trimStart().trimEnd();
-      // Remove quotation marks from string since not apart of the CSS or HTML
-      const nodeTextWithoutQuotes = trimmedNodeText.substring(1, trimmedNodeText.length - 1);
-      const migratedText = migration.migrate(nodeTextWithoutQuotes);
-      const migratedTextLines = migratedText.split('\n');
-      const isMultiline = migratedTextLines.length > 1;
-
-      // If migrated text is now multiline, update quotes to avoid compilation
-      // errors
-      if (isMultiline) {
-        nodeText = nodeText.replace(trimmedNodeText, '`' + nodeTextWithoutQuotes + '`');
-      }
-
-      this._printAndUpdateNode(
-        node.getSourceFile(),
-        node,
-        ts.factory.createRegularExpressionLiteral(
-          nodeText.replace(
-            nodeTextWithoutQuotes,
-            migratedTextLines
-              .map((line: string, index: number) => {
-                // Only need to worry about indentation when adding new lines
-                if (isMultiline && index !== 0 && line != '\n') {
-                  const leadingWidth = node.getLeadingTriviaWidth();
-                  if (leadingWidth > 0) {
-                    line = ' '.repeat(leadingWidth - 1) + line;
-                  }
-                }
-                return line;
-              })
-              .join('\n'),
-          ),
-        ),
-      );
+    let migratedText = migration.migrate(node.text);
+    let migratedTextLines = migratedText.split('\n');
+
+    // Update quotes based on if its multiline or not to avoid compilation errors
+    if (migratedTextLines.length > 1) {
+      // Add correct identation for new lines before replacing
+      migratedText = migratedTextLines
+        .map((line: string, index: number) => {
+          if (index !== 0 && line != '\n') {
+            const leadingWidth = node.getLeadingTriviaWidth();
+            if (leadingWidth > 0) {
+              line = ' '.repeat(leadingWidth - 1) + line;
+            }
+          }
+          return line;
+        })
+        .join('\n');
+
+      migratedText = '`' + migratedText + '`';
+    } else {
+      // Need to grab quotation because it is not included in node.text
+      const quotation = node.getText().trimStart()[0];
+      migratedText = quotation + migratedText + quotation;
     }
+
+    this._printAndUpdateNode(
+      node.getSourceFile(),
+      node,
+      ts.factory.createRegularExpressionLiteral(migratedText),
+    );
   }
 
-  private _migrateModuleSpecifier(specifierLiteral: ts.StringLiteralLike) {
+  private _migrateModuleSpecifier(
+    specifierLiteral: ts.StringLiteralLike,
+    importClause?: ts.ImportClause,
+  ) {
     const sourceFile = specifierLiteral.getSourceFile();
 
     // Iterate through all activated migrators and check if the import can be migrated.
     for (const migrator of this.upgradeData) {
+      if (importClause) {
+        importClause.namedBindings?.forEachChild(childNode => {
```

Comment:
Optional future nit: Even here you could just use `namedBindings.elements` instead of `forEachChild`. Then you also get proper type completion and could drop the `.kind` check since `NamedImports` elements are guaranteed import specifiers (and this is guaranteed per AST and types)

Reply:
awesome, thanks!